src/lib.rs

```diff
@@ -385,6 +394,73 @@ impl Interpolate for Quaternion<f32> {
   }
 }
 
+#[cfg(feature = "impl-nalgebra")]
+impl<N : Scalar, D : DimName> Interpolate for na::Point<N, D> 
+  where DefaultAllocator : Allocator<N, D>,
+        <DefaultAllocator as Allocator<N, D>>::Buffer: Copy,
+        N : Interpolate,
+{
+  fn lerp(a: Self, b: Self, t: f32) -> Self {
+    let lerp = |c1 : N, c2 : N| { Interpolate::lerp(c1, c2, t) };
+    let coords = na::Matrix::zip_map(&a.coords, &b.coords, lerp);
+    na::Point::from_coordinates(coords)
+  }
+}
+
+#[cfg(feature = "impl-nalgebra")]
+impl Interpolate for na::Vector1<f32>
+{
+  fn lerp(a: Self, b: Self, t: f32) -> Self 
+  {
+    na::Matrix::zip_map(&a, &b, |c1, c2| Interpolate::lerp(c1, c2, t))
```

Comment:
See, here you’re using the closure directly, and no type ascriptions needed. :smirk: